### PR TITLE
[CSL-1972] Do a single DB and mempool read when computing wallet balance

### PR DESCRIPTION
**Pull request by Martoon-00**
Rebased #1994

### CODE REVIEW

File: node/src/Pos/Client/Txp/Balances.hs [hunk 1]
```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE TypeFamilies #-}
+
+module Pos.Client.Txp.Balances
```
Comment (gromakovsky) [inline node/src/Pos/Client/Txp/Balances.hs]: What? Why did you add this module here? Probably by mistake.

**Comment by georgeee** [comment]
@Martoon-00 @gromakovsky @adinapoli-iohk shall we close it?

**Comment by adinapoli-iohk** [comment]
@georgeee I think so.

@nc6 's [current work](https://github.com/input-output-hk/cardano-sl/pull/2394) will address this and many other concerns.
